aptos-move/move-examples/token_objects/ambassador/client/ambassador.ts

```diff
@@ -0,0 +1,144 @@
+// Copyright © Aptos Foundation
+// SPDX-License-Identifier: Apache-2.0
+
+import { AptosAccount, AptosClient, HexString, Provider, Network, Types } from "aptos";
+
+const NODE_URL = process.env.APTOS_NODE_URL || "https://fullnode.testnet.aptoslabs.com";
+const provider = new Provider(Network.TESTNET);
+
+async function getTokenAddr(ownerAddr: HexString, tokenName: string): Promise<HexString> {
+    const tokenOwnership = await provider.getOwnedTokens(ownerAddr)
+    for (const ownership of tokenOwnership.current_token_ownerships_v2) {
+        if (ownership.current_token_data.token_name === tokenName) {
+            return new HexString(ownership.current_token_data.token_data_id);
+        }
+    }
+    console.log(`Token ${tokenName} not found`);
+    process.exit(1);
+}
+
+async function waitForEnter() {
+    return new Promise<void>((resolve, reject) => {
+        const rl = require("readline").createInterface({
+            input: process.stdin,
+            output: process.stdout
+        });
+
+        rl.question('Please press the Enter key to proceed ...\n', () => {
+            rl.close();
+            resolve();
+        });
+    });
+}
+
+class AmbassadorClient extends AptosClient {
```

Comment:
why do you need to extend `AptosClient`? I think you dont need it since you use Provider class.

Reply:
Removed.